### PR TITLE
Fix i18n file diff test

### PR DESCRIPTION
**Pull request by rebeccaalpert**
The diff test for i18n files was still using the old file location (`./locales`). I updated it to handle public and packages locales folders.

### CODE REVIEW

File: test-frontend.sh
Comment (spadgett) [inline test-frontend.sh:24]: I believe this can be one command.

```suggestion
GIT_STATUS=[$(git status --short --untracked-files -- public/locales packages/**/locales)]
```

Comment (rebeccaalpert) [reply test-frontend.sh]: All set!

**Review by spadgett — APPROVED** [review]
/lgtm

Thanks!

**Comment by rebeccaalpert** [comment]
Frontend test is going to fail until https://github.com/openshift/console/pull/7185 merges.

**Comment by spadgett** [comment]
Looks like it's working :)

/hold until #7185 merges to fix the JSON files

**Comment by rebeccaalpert** [comment]
/retest

#7185 merged.

**Comment by rebeccaalpert** [comment]
Rebasing.

**Comment by rebeccaalpert** [comment]
Another JSON file came in; updated to see if frontend tests will now pass.

**Comment by rebeccaalpert** [comment]
Another JSON file wasn't checked in.

File: frontend/public/i18next-parser.config.js
Comment (spadgett) [inline frontend/public/i18next-parser.config.js:5]: Oh, we shouldn't be referencing variables in the parser config in the app. I think we need to just make the locales map a separate variable in the language selector modal.

**Comment by rebeccaalpert** [comment]
One of the i18n files that hadn't been checked in was conflicting with a namespace in another package. I updated the namespace that was smaller to fix the e2e tests. Long term, we should have a tool to check for these. Can you review this @spadgett assuming tests now pass? Thanks.

File: frontend/public/i18next-parser.config.js
Comment (spadgett) [inline frontend/public/i18next-parser.config.js:1]: This file shouldn't be in the `public` folder if it's just the parser config. `public` is for code that is part of the app itself.

**Review by spadgett — CHANGES_REQUESTED** [review]

File: frontend/public/i18next-parser.config.js
Comment (rebeccaalpert) [reply frontend/public/i18next-parser.config.js:5]: @spadgett - We're also using it in webpack.config.ts - Joachim had it set up to process Chinese a certain way when it's available. Are you ok with maintaining it in two places?

File: frontend/public/i18next-parser.config.js
Comment (rebeccaalpert) [reply frontend/public/i18next-parser.config.js:1]: I'll try to move it. Fingers crossed it builds.

**Comment by rebeccaalpert** [comment]
@spadgett - I updated file locations and imports; still have an open query on the support locales object.

**Comment by rebeccaalpert** [comment]
```
Failed Specs:
1) Modal Annotations : Delete Annotation
   StaleElementReferenceError: stale element reference: element is not attached to the page document
```

/retest

File: frontend/i18next-parser.config.js
Comment (spadgett) [inline frontend/i18next-parser.config.js:57]: If we only need the keys, no reason for this to be a map with display names.

File: frontend/public/components/modals/language-preferences-modal.tsx
Comment (spadgett) [inline frontend/public/components/modals/language-preferences-modal.tsx:20]: Instead of using a map and transforming it, just initial the variable in the format we need.

File: frontend/webpack.config.ts
Comment (spadgett) [inline frontend/webpack.config.ts:26]: Don't initial this as a map and format later. Better to just specify it inline in the format we need.

File: frontend/i18next-parser.config.js
Comment (rebeccaalpert) [reply frontend/i18next-parser.config.js:57]: Good point, thanks!

File: frontend/public/components/modals/language-preferences-modal.tsx
Comment (rebeccaalpert) [reply frontend/public/components/modals/language-preferences-modal.tsx:20]: We're also using supportedLocales as-is to fill the dropdown on line 45 though.

File: frontend/webpack.config.ts
Comment (spadgett) [inline frontend/webpack.config.ts:209]:
```suggestion
      localesToKeep: ['en', 'ja'],
```

And when we get Chinese, we can just add `zh-cn` to the array without needing `map`

**Review by spadgett — APPROVED** [review]
/lgtm

**Pull request merged**
